src/ORM/Association/BelongsToMany.php

```diff
@@ -300,6 +300,29 @@ public function junction(Table|string|null $table = null): Table
         return $this->_junctionTable = $table;
     }
 
+    /**
+     * Set the junction property name.
+     *
+     * @param string $junctionProperty Property name.
+     * @return $this
+     */
+    public function setJunctionProperty(string $junctionProperty)
```

Comment:
I wonder if we should rather stay consistent on the naming here in regards to the fields.
The "property" is the implementation detail, isnt it?
They might be implemented using class properties, but in the end those are all entity fields I guess.

So maybe we should use
- junction name
- junction key
- junction field

or alike instead?

Reply:
Well, we do have the config key `propertyName` and `setProperty()/getProperty()` methods. So in that regard `junctionProperty` is consistent.

Reply:
Users could set the junction property to valid field name, right?

Reply:
We could continue to require the name to have a leading underscore.
Avoids them shooting themselves in the foot I assume

Reply:
> Users could set the junction property to valid field name, right?

Yes they could.

> We could continue to require the name to have a leading underscore.

That would defeat the purpose of this feature. Users would again have to resort to re-keys the values if they don't want the underscore.

The same risk exists if one customizes`propertyName` too, so users just need to be mindful of the name they choose.

Reply:
That's true. Seems fine to me.